webapp/src/Service/SubmissionService.php

```diff
@@ -530,8 +530,9 @@ public function submitSolution(
             if ($source !== 'shadowing' && $language->getFilterCompilerFiles()) {
                 $matchesExtension = false;
                 foreach ($language->getExtensions() as $extension) {
-                    $extensionLength = strlen($extension);
-                    if (substr($file->getClientOriginalName(), -$extensionLength) === $extension) {
+                    $extensionWithDot = '.' . $extension;
+                    $extensionLength = strlen($extensionWithDot);
+                    if (substr($file->getClientOriginalName(), -$extensionLength) === $extensionWithDot) {
```

Comment:
nit: use `str_ends_with` here to make the intent more clear?

Reply:
Makes sense, I guess it wasn't used before since the function is new in PHP 8.0.